### PR TITLE
Adds POST Goods Out Note Event and customFields params to Order GET

### PR DESCRIPTION
**Pull request by charan41594**
## PR Type
What kind of change does this PR introduce?
```
[ ] Bugfix
[X] Feature
[ ] Code style update (formatting, local variables)
[X] Refactoring (no functional changes, no api changes)
[ ] Build related changes
[ ] CI related changes
[ ] Documentation content changes
[ ] Tests
[ ] Other
```

## Checklist
Have you done the following?
```
[X] Exported any new types in index.ts
[ ] Merged master
[ ] Run `npm run lint`
[ ] Ensure the tests pass
```

### CODE REVIEW

File: src/BrightpearlClient.ts [hunk 1]
```diff
@@ -49,8 +50,9 @@ class BrightpearlClient {
         return httpResponse.data;
     }
 
-    public async getOrder(orderId: number): Promise<OrderInformation[]> {
-        const httpResponse: AxiosResponse<Response<OrderInformation[]>> = await this.axios.get(`${this.baseUrl}/${ENDPOINTS.order.get}/${orderId}`,
+    public async getOrder(orderId: number, customFields: any): Promise<OrderInformation[]> {
```
Comment (carlrc) [inline src/BrightpearlClient.ts]: `customFields` should be made optional

Comment (carlrc) [reply src/BrightpearlClient.ts]: Axios handles URL params natively. Would be good to use vs doing it our self. https://github.com/axios/axios

File: src/IBrightpearlClient.ts [hunk 2]
```diff
@@ -842,6 +842,16 @@ interface ProductChannel {
     integrationDetail?: ProductChannelIntegrationDetail;
 }
 
+interface GoodsOutNote {
```
Comment (carlrc) [inline src/IBrightpearlClient.ts]: Isn't there already a GoodsOutNote type defined? Does linting succeed?

File: src/Endpoints.ts [hunk 3]
```diff
@@ -22,6 +22,9 @@ const ENDPOINTS  = Object.freeze({
         goods_note: {
             search: `${WAREHOUSE}/goods-note/goods-out-search`
         },
+        goods_out_note: {
```
Comment (carlrc) [inline src/Endpoints.ts]: I think the goods_note endpoint is a mistake and should be changed to goods_out_note and include your new endpoint

File: src/IBrightpearlClient.ts [hunk 4]
```diff
@@ -842,6 +842,16 @@ interface ProductChannel {
...
+    eventCode: string;
```
Comment (carlrc) [inline src/IBrightpearlClient.ts]: Can be made an ENUM with available values http://api-docs.brightpearl.com/warehouse/goods-out-note%20event/post.html

File: src/IBrightpearlClient.ts (hunk 2 above)
Comment (carlrc) [reply src/IBrightpearlClient.ts]: I think the name should reflect that it is an event for a Goods Out Note, not a Goods Out Note itself

File: src/IBrightpearlClient.ts [hunk 5]
```diff
@@ -842,6 +842,16 @@ interface ProductChannel {
...
+    occured: string;
+    eventOwnerId: number;
+}
+
+interface GoodsOutNoteEvent {
+    events: GoodsOutNote[];
```
Comment (carlrc) [inline src/IBrightpearlClient.ts]: All new types should be exported in index.ts

File: src/BrightpearlClient.ts [hunk 6]
```diff
@@ -249,6 +251,13 @@ class BrightpearlClient {
         return httpResponse.data.response;
     }
 
+    public async postGoodsOutNoteEventShipped(gonId: number, goodsOutNoteEvent: GoodsOutNoteEvent): Promise<number> {
```
Comment (carlrc) [inline src/BrightpearlClient.ts]: I think it makes more sense to name it just `postGoodsOutNoteEvent`, as they could pass in many types of event codes

Comment (carlrc) [reply src/BrightpearlClient.ts]: I think it's better to avoid acronyms, as we are writing out the params fully everywhere else. So `GoodsOutNoteId`

Comment (carlrc) [reply src/BrightpearlClient.ts]: Also, the API docs say that an empty object is returned with a valid request not a number. So I'd either return `void` or just return the `object`

Comment (charan41594) [reply src/BrightpearlClient.ts]: Hi Carl, made changes are advised.

File: src/Endpoints.ts [hunk 7]
```diff
@@ -39,7 +40,6 @@ const ENDPOINTS  = Object.freeze({
             put: `${ORDER}/order`
         },
         custom_field: {
-            get: `${ORDER}/order`,
```
Comment (carlrc) [inline src/Endpoints.ts]: Was this an unused endpoint?

File: src/BrightpearlClient.ts [hunk 8]
```diff
@@ -49,8 +50,9 @@ class BrightpearlClient {
         return httpResponse.data;
     }
 
-    public async getOrder(orderId: number): Promise<OrderInformation[]> {
-        const httpResponse: AxiosResponse<Response<OrderInformation[]>> = await this.axios.get(`${this.baseUrl}/${ENDPOINTS.order.get}/${orderId}`,
+    public async getOrder(orderId: number, customFields?: any): Promise<OrderInformation[]> {
```
Comment (carlrc) [inline src/BrightpearlClient.ts]: Should be type string, given we are working with URLs? Or is there a specific benefit to using `any`?

Comment (charan41594) [reply src/BrightpearlClient.ts]: updated in the next commit

File: src/Endpoints.ts (hunk 7 above)
Comment (charan41594) [reply src/Endpoints.ts]: yes carl - this was unused and as of now this URL is exact duplicate of order.get(). That's thee reason behind removing this. So do you want to keep it separate?

**Pull request closed without merging**